src/tests/test_telemetry.py

```diff
@@ -0,0 +1,76 @@
+from pathlib import Path
+from unittest.mock import ANY, Mock
+import pytest
+import urllib.request
+import duckdb
+from sql.telemetry import telemetry
+from sql import plot
+
+
+@pytest.fixture
+def simple_db_conn():
+    if not Path("iris.csv").is_file():
+        urllib.request.urlretrieve(
+            "https://raw.githubusercontent.com/plotly/datasets/master/iris-data.csv",
+            "iris.csv",
+        )
+    conn = duckdb.connect(database=":memory:")
+    return conn
+
+
+@pytest.fixture
+def mock_log_api(monkeypatch):
+    mock_log_api = Mock()
+    monkeypatch.setattr(telemetry, "log_api", mock_log_api)
+    yield mock_log_api
+
+
+def test_boxplot_telemetry_execution(mock_log_api, simple_db_conn):
```

Comment:
We need to use a tmp folder here since `simple_db_conn` is writing a file locally (so it won't affect other tests)

Reply:
I tried to create the file into the tmp folder, but I faced the problem that the SQL query was not able to read the file with any folder name (failed test case: `test_data_frame_telemetry_execution`)

e.g.
1. The current folder has: iris.csv
=> `%sql SELECT * FROM iris.csv`, works fine ✅

2. The current folder has: tmp/iris.csv
=> `%sql SELECT * FROM tmp/iris.csv`, failed ❌

Probably for this test case I need to find another workaround

Reply:
Use the python console, and `os.listdir()` to see it's content.
You'd need to select via an engine before querying.

Reply:
I saw a similar test in test_plot, we should make a fixture out of it and consume in both places.